src/app/fact-page/fact-page.component.ts

```diff
@@ -25,22 +25,23 @@ export class FactPageComponent implements OnInit, OnDestroy {
 
   public ngOnInit() {
     this.route.params
-      .pipe(takeUntil(this.onDestroy$))
-      .subscribe(params => {
-        if (isNaN(params.id)) {
-          this.router.navigateByUrl('/');
-        } else {
-          this.service.getFact(params.id)
-            .pipe(takeUntil(this.onDestroy$))
-            .subscribe((response: ApiResponse) => {
-              this.numberFact = response;
-              this.processing = false;
-            }, (error: HttpErrorResponse) => {
-              this.networkError = error;
-              this.processing = false;
-            });
-        }
-      });
+      .pipe(
+        takeUntil(this.onDestroy$),
+        map(params => {
+          if (isNaN(params.id)) {
+            this.router.navigateByUrl('/');
+          } else {
+            this.service.getFact(params.id)
+              .subscribe((response: ApiResponse) => {
```

Comment:
Please consider to remove `subscribe` from `map`, as it supposed to be clear function without side effects.